great_expectations/data_context/types/base.py

```diff
@@ -97,6 +97,9 @@ def from_commented_map(cls, commented_map: CommentedMap):
                 commented_map
             )
             if isinstance(config, dict):
+                if config.get("class_name") == "RuleBasedProfiler":
+                    config.pop("class_name", None)
+                    config.pop("module_name", None)
```

Comment:
Ugh this is ugly - how do I differentiate between subclasses that want these attrs and those that don't?